src/psyclone/tests/psyGen_test.py

```diff
@@ -4199,9 +4199,24 @@ def test_fparser2astprocessor_process_declarations_stmt_functions(
     '''
     from fparser.common.readfortran import FortranStringReader
     from fparser.two.Fortran2003 import Specification_Part
+    from fparser.two.Fortran2003 import Stmt_Function_Stmt
     fake_parent = KernelSchedule("dummy_schedule")
     processor = Fparser2ASTProcessor()
 
+    # If 'a' is not declared it could be a statement function, which are
+    # unsupported and produce a NotImplementedError.
+    reader = FortranStringReader("a(x) = 1")
+    fparser2spec = Stmt_Function_Stmt(reader)
```

Comment:
I think you now have this part of the test duplicated at line 4222 below.

Reply:
They are similar but 4209 uses Stmt_Function_Stmt() and 4222 uses Specification_Part(), the idea is that when fparser/171 is fixed we still want the first to be unsupported but we may want to delete from 4216 to the end of the test or give it a different functionality depending of the output that parsing this expression will return.

I updated the comment to make it clearer.

Reply:
Great.